SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -240,6 +248,28 @@ void USpatialReceiver::OnRemoveEntity(const Worker_RemoveEntityOp& Op)
 
 	RemoveActor(Op.entity_id);
 	OnEntityRemovedDelegate.Broadcast(Op.entity_id);
+
+	if (NetDriver->IsServer())
+	{
+		if (FString* WorkerName = WorkerConnectionEntity.Find(Op.entity_id))
+		{
+			TWeakObjectPtr<USpatialNetConnection> PlayerConnectionPtr = NetDriver->FindWorkerConnectionFromWorkerId(*WorkerName);
+			if (USpatialNetConnection* PlayerConnection = PlayerConnectionPtr.Get())
+			{
+				if (APlayerController* Controller = PlayerConnection->GetPlayerController(nullptr))
+				{
+					Worker_EntityId PCEntity = PackageMap->GetEntityIdFromObject(Controller);
+					if (AuthorityPlayerControllerConnectionMap.Find(PCEntity))
+					{
+						UE_LOG(LogSpatialReceiver, Verbose, TEXT("Worker %s disconnected after its system identity was removed."), *(*WorkerName));
+						PlayerConnection->CleanUp();
```

Comment:
Can we centralize the code here and in `USpatialReceiver::OnHeartbeatComponentUpdate` when the client dcs?

Reply:
There is not much cpde that can be centralized atm (2 lines), and eventually we are going to remove the Heartbeat component way, so not sure it is worth creating a new method.

Reply:
I think it's worth doing